warehouse/integrations/middleware/sqlquerywrapper/tx.go

```diff
@@ -0,0 +1,33 @@
+package sqlquerywrapper
+
+import (
+	"context"
+	"fmt"
+	"time"
+)
+
+// txWithTimeout is a wrapper around sql.Tx that adds a timeout to Rollback.
+type txWithTimeout struct {
+	*Tx
+	timeout time.Duration
+}
+
+func (tx *txWithTimeout) Rollback() error {
+	ctx, cancel := context.WithTimeout(context.Background(), tx.timeout)
+	defer cancel()
+
+	errCh := make(chan error, 1)
+	go func() {
+		errCh <- tx.Tx.Rollback()
+	}()
```

Comment:
For customers, we have seen queries getting stuck at Rollback for couple of hours. 
https://rudderlabs.slack.com/archives/C02AALVFHJN/p1649286005516699?thread_ts=1649079527.127459&cid=C02AALVFHJN

Reply:
@achettyiitr are we sure that passing a context with a timeout to `BeginTx` doesn't timeout also the rollback in case it takes too long?

Reply:
We could test this with a `SELECT pg_sleep(X)` where `X` would be a number of seconds greater than the ones passed to the context timeout.

Reply:
FYI, some of these could return `pq: canceling statement due to user request` instead of `context.Canceled` or `context.DeadlineExceeded`. It might be worth checking for it (e.g. in case you got some retry policy in place in case that would affect the retry logic). Feel free to ignore if you think it won't affect the logic in any way.